### PR TITLE
[PLT-6639] Tests for Form

### PR DESCRIPTION
**Pull request by imogenem**
## Description
Tests for Form

## Notes
Tried testing the errorScroll and enter props with mock & spy functions, but could not find a test that worked with functional components.

## Where to Start
Form.test.jsx

### CODE REVIEW

File: packages/riipen-ui/src/components/Form.test.jsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    it("appends higher order values to classes prop with withClass decorator", () => {
+      const classes = ["riipen", "riipen-form"];
+
+      const wrapper = mount(<Form />);
+
+      expect(wrapper.find("Form").props().classes).toEqual(classes.sort());
+    });
+  });
+
+  describe("error prop", () => {
```
Comment (licarm) [inline packages/riipen-ui/src/components/Form.test.jsx:69]: Can we check that the error gets displayed using `.containsMatchingElement()` or something similar? And maybe be check that it gets put inside a `Typography` element if it is a string.

File: packages/riipen-ui/src/components/Form.test.jsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      expect(
+        wrapper
+          .find("Form")
+          .childAt(0)
+          .hasClass("errorContainer")
+      ).toEqual(true);
+    });
+  });
+
+  describe("errors prop", () => {
```
Comment (licarm) [inline packages/riipen-ui/src/components/Form.test.jsx:107]: Same as above. Also wondering if we can check that `renderErrorsObject` processes errors correctly when given as an object, or object inside an object.

**Review by licarm — CHANGES_REQUESTED** [review]

File: packages/riipen-ui/src/components/Form.test.jsx (hunk 1 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Form.test.jsx:69]: Added both plus a test for `renderErrorNode`

File: packages/riipen-ui/src/components/Form.test.jsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      ).toEqual(errors[1]);
+    });
+
+    it("renders errors within Typography element if errors are strings", () => {
+      const errors = ["Error", "Sad"];
+
+      const wrapper = mount(<Form errors={errors} />);
+
+      expect(wrapper.find("Typography").contains(errors[0])).toEqual(true);
+    });
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Form.test.jsx:217]: I feel like these are testing the same thing? Do we need both tests?

**Review by jordanell — CHANGES_REQUESTED** [review]
Just one simplification. Looks great!

File: packages/riipen-ui/src/components/Form.test.jsx (hunk 3 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Form.test.jsx]: I'll take it out, it's meant to test that an error is rendered within a Typography element, but that's already covered in the error prop block.

**Review by jordanell — APPROVED** [review]

**Review by licarm — APPROVED** [review]

**Pull request merged**